### PR TITLE
Add is_nft to mojom::BlockchainToken

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/26395

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/asset_ratio_response_parser.cc [hunk 1]
```diff
@@ -288,7 +288,7 @@ mojom::BlockchainTokenPtr ParseTokenInfo(const std::string& json,
 
   return mojom::BlockchainToken::New(
       eth_addr.ToChecksumAddress(), *name, "" /* logo */, is_erc20, is_erc721,
-      *symbol, decimals, true, "", "", chain_id, coin);
+      is_erc721 /* is_nft */, *symbol, decimals, true, "", "", chain_id, coin);
```
Comment (yrliou) [inline components/brave_wallet/browser/asset_ratio_response_parser.cc:291]: This is used by `AssetRatioService::OnGetTokenInfo` only which is ETH only.

File: components/brave_wallet/browser/blockchain_list_parser.cc [hunk 2]
```diff
@@ -99,6 +99,8 @@ bool ParseTokenList(const std::string& json,
     else
       blockchain_token->is_erc721 = false;
 
+    blockchain_token->is_nft = blockchain_token->is_erc721;
```
Comment (yrliou) [inline components/brave_wallet/browser/blockchain_list_parser.cc:102]: We don't have is_nft used by our token maps from wallet component extension atm.

File: components/brave_wallet/common/eth_request_helper.cc [hunk 3]
```diff
@@ -648,7 +648,7 @@ bool ParseWalletWatchAssetParams(const std::string& json,
   }
 
   *token = mojom::BlockchainToken::New(
-      eth_addr.ToChecksumAddress(), *symbol /* name */, logo, true, false,
+      eth_addr.ToChecksumAddress(), *symbol /* name */, logo, true, false, false,
```
Comment (yrliou) [inline components/brave_wallet/common/eth_request_helper.cc]: This function is for ERC20 only.

File: android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java [hunk 4]
```diff
@@ -964,18 +964,6 @@ public static String getTimeframeString(int assetPriceTimeframe) {
         }
     }
 
-    public static BlockchainToken createEthereumBlockchainToken(String chainId) {
```
Comment (yrliou) [inline android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java:967]: This function seems to be unused.

**Review by darkdh — APPROVED** [review]
backend lgtm

**Review by Douglashdaniel — APPROVED** [review]
++ Front-end!

**Review by SergeyZhukovsky — APPROVED** [review]
++ Android part

**Pull request merged**